app/models/mixins/live_metrics_mixin.rb

```diff
@@ -0,0 +1,39 @@
+module LiveMetricsMixin
+  extend ActiveSupport::Concern
+
+  class MetricValidationError < RuntimeError; end
+
+  included do
+    def metrics_available
+      raise 'LiveMetricsMixing: metrics_available missing'
+    end
+
+    def metric(_metric, _start_time, _end_time, _interval)
+      raise 'LiveMetricsMixing: metric method missing'
+    end
+
+    def metrics(*metrics, start_time, end_time, interval)
+      metrics = metrics_available if metrics.empty?
+      processed = Hash.new { |h, k| h[k] = {} }
+      metrics.each do |m|
+        values = metric(m, start_time, end_time, interval)
+        values.each do |k, a|
+          processed[k].merge!(a)
+        end
+      end
+      processed
+    end
+
+    def region_id
+      # Needed in the charts scenario
+      # TODO This should be at MiddlewareServer level ?
+      1
+    end
+
+    def first_and_last_capture(interval_name = "hourly")
+      # Mainly for compatibility with current Metrics model
+      # but not sure if it is needed from a pure LiveMetrics perspective
+      raise 'LiveMetricsMixin: first_and_last_capture missing'
+    end
```

Comment:
I think these methods were written coming from a Java perspective.
In Java the concept of interface sometimes is used to show that a LiveMetricsMixin should have a set of methods as part of the business contract.
Is it a similar way in RoR to declare same concept ?
I know that this is out of the PR review and it is more a Java -> Ruby general topic, but just in case :)